tests/units/Toolbox.php

```diff
@@ -891,4 +891,23 @@ public function testIsAPIDeprecated(string $class, bool $is_deprecated) {
       $this->boolean(\Toolbox::isAPIDeprecated($class))->isEqualTo($is_deprecated);
    }
 
+   protected function urlProvider() {
+      return [
+         ['http://localhost', true],
+         ['https://localhost', true],
+         ['https;//localhost', false],
+         ['https://glpi-project.org', true],
+         ['https://glpi+project-org', true],
```

Comment:
true for this one ?

Reply:
I'm not expert in URLs, but "localhost" is valid and all characters of "glpi+project-org" are allowed, so yes...

Reply:
`+` sign is not valid in the regex (in this regex, + are only used for matches count), and seems not valid in hosts/domains names (see https://stackoverflow.com/questions/8972397/plus-sign-on-subdomains-names ).

This one is only valid due to the fact that the regex does not check end of string, so `https://glpi` validates the regex.

I proposed a change on regex to check the whole string in order to prevent false positives.

```suggestion
         ['https://glpi+project-org', false],
```